### PR TITLE
feat: learner credit management activity tab empty states

### PR DESCRIPTION
**Pull request by adamstankiewicz**
# Description

Empty states for the Learner Credit Management budget detail page route's "Activity" tab empty states.

### NO content assignments and NO spent transactions

image

### NO content assignments and HAS spent transactions

image

### HAS content assignments and NO spent transactions

image

### HAS content assignments and HAS spent transactions

image

# For all changes

- [x] Ensure adequate tests are in place (or reviewed existing tests cover changes)

# Only if submitting a visual change

- [x] Ensure to attach screenshots
- [ ] Ensure to have UX team confirm screenshots

### CODE REVIEW

File: src/components/learner-credit-management/data/utils.js [hunk 1]
```diff
@@ -185,6 +190,101 @@ export const orderOffers = (offers) => {
...
+ * @returns An object containing the first page of spent transactions and (if applicable) content assignments.
+ */
+export async function retrieveBudgetDetailActivityOverview({
+  queryKey,
+  subsidyAccessPolicy,
+  enterpriseUUID,
+  isTopDownAssignmentEnabled,
+}) {
+  const isBudgetAssignable = !!(isTopDownAssignmentEnabled && subsidyAccessPolicy?.isAssignable);
+  const budgetId = queryKey[4];
```
Comment (iloveagent57) [inline src/components/learner-credit-management/data/utils.js:271]: nit: this function only depends on the last element of the `queryKey` array (the budget id) - would it be better to explicitly expected `budgetId` as an argument provided by the caller?

**Review by iloveagent57 — APPROVED** [review]
This looks great

File: src/components/learner-credit-management/data/utils.js (hunk 1 above)
Comment (adamstankiewicz) [reply src/components/learner-credit-management/data/utils.js]: Updated!

File: docs/decisions/0006-tanstack-react-query.rst [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+  queryClient.invalidateQueries({
+    queryKey: learnerCreditManagementQueryKeys.budgetActivityOverview(budgetId),
+  })
+
+The recommendation for new asynchronous network calls moving forward is to use ``@tanstack/react-query``. Additionally, it's recommended to incrementally migrate existing network calls to use ``@tanstack/react-query``. This will allow us to avoid a large refactoring effort and instead migrate to the library over time as we touch existing network calls.
+
+Consequences
+************
+
+The entire application is wrapped within ``QueryClientProvider``, which contains a default configuration to at least extend the ``staleTime`` to be 20 seconds. This change in default behavior is to enable the use of ``@tanstack/react-query`` as a state management. Instead of queries becoming instantly stale after success (i.e., ``staleTime: 0``) where network calls would be re-fetched on all window refocuses and component mounts, etc., having a ``staleTime`` of 20 seconds will keep the data fresh for 20 seconds before the query becomes stale, preventing unnecessary API calls (e.g., when calling duplicate ``useQuery`` hooks in the same rendering lifecycle).
```
Comment (iloveagent57) [inline docs/decisions/0006-tanstack-react-query.rst:64]: nit: "as a state management _tool_"? or "state manager"

**Review by iloveagent57 — APPROVED** [review]

File: docs/decisions/0006-tanstack-react-query.rst (hunk 2 above)
Comment (adamstankiewicz) [reply docs/decisions/0006-tanstack-react-query.rst:64]: I think either could work? I'm using "state manager" based on https://tkdodo.eu/blog/react-query-as-a-state-manager.

Comment (adamstankiewicz) [reply docs/decisions/0006-tanstack-react-query.rst:64]: Fixed the typo. Good catch!

**Pull request merged**